### DISCUSSION TITLE
Extract text in a proper order from multiple line

### DISCUSSION THREAD

**Discussion by ervinwirth** [jsvine/pdfplumber#1158] [Ask for help with specific PDFs]
I have a PDF, where the Consumption address ('Felhasználási hely címe' in Hungarian) written in multiple lines; so the extraction - because of coordinate flow - is not in the right order:

![image](https://github.com/jsvine/pdfplumber/assets/90049125/c9c1e824-6682-427f-b4d8-e97b7b698c1e)

The output text:

_xxxxx ; 1881 Budapest Csontváry Kosztka
Felhasználási hely címe:
Tivadar utca xxxxx_

The output I would like:

_Felhasználási hely címe:
xxxxx ; 1881 Budapest Csontváry Kosztka
Tivadar utca xxxxx_

I use this method:
text_all_pages += page.dedupe_chars().extract_text(x_tolerance=1)

Any idea to solve this?

**Comment by jsvine** [comment]
Can you share the (redacted) PDF? It will be difficult to suggest a solution without that. In general, though, the output is what I would expect, since `1881 Budapest Csontváry Kosztka` is higher on the page than `Felhasználási hely címe:`. If you want to process the lines in a different order, that may require some custom Python logic re. line positions. For that, you might want to try `page. extract_text_lines(...)` or `page. extract_words(...)`.

**Comment by ervinwirth** [comment]
Of course,
I hope I added it.
[redacted_pdf_example.pdf](https://github.com/user-attachments/files/15972787/redacted_pdf_example.pdf)

**Reply by victornw** [reply]
did you find any solutions? i've be dealing with the same problem

**Reply by jsvine** [reply]
Thank you, that's helpful. Looking at that PDF, it looks like the general suggestion I've made above still applies. The text is being extracted correctly, but you'll need to write a bit of your own logic to determine the output order, since > since `1881 Budapest Csontváry Kosztka` is higher on the page than `Felhasználási hely címe:`.

There are a few ways you could do this, but the best solution for you will depend on the particular patterns you expect to see in other such examples.
